python/monarch/_src/job/skypilot.py

```diff
@@ -0,0 +1,488 @@
+
+# Install Rust
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+source $HOME/.cargo/env
+rustup default nightly
+
+# Install Python dependencies and build Monarch from source
+cd ~/sky_workdir
+pip install setuptools-rust maturin
+pip install -r torch-requirements.txt -r build-requirements.txt
+CC=gcc-11 CXX=g++-11 USE_TENSOR_ENGINE=0 pip install --no-build-isolation .
+
+echo "Done installing Monarch"
+"""
+
+
+def _configure_transport() -> None:
+    """Configure the Monarch transport. Deferred import to avoid import errors."""
+    from monarch._rust_bindings.monarch_hyperactor.channel import ChannelTransport
+    from monarch._rust_bindings.monarch_hyperactor.config import configure
+
+    configure(default_transport=ChannelTransport.TcpWithHostname)
+
+
+def _attach_to_workers_wrapper(name: str, ca: str, workers: List[str]):
+    """Wrapper around attach_to_workers with deferred import."""
+    from monarch._src.actor.bootstrap import attach_to_workers
+
+    return attach_to_workers(name=name, ca=ca, workers=workers)
+
+
+class SkyPilotJob(JobTrait):
+    
+    """
+    A job scheduler that uses SkyPilot to provision cloud instances.
+
+    SkyPilot supports multiple cloud providers (AWS, GCP, Azure, Lambda, etc.)
+    and can automatically select the cheapest available option.
+
+    This implementation:
+    1. Uses sky.launch() to provision cloud instances with specified resources
+    2. Runs Monarch workers on each node via a startup script
+    3. Connects to workers using their IP addresses from the cluster handle
+
+    Example:
+        >>> import sky
+        >>> from monarch.job import SkyPilotJob
+        >>>
+        >>> job = SkyPilotJob(
+        ...     meshes={"trainers": 2},
+        ...     resources=sky.Resources(accelerators="A100:1"),
+        ...     cluster_name="my-monarch-cluster",
+        ... )
+        >>> state = job.state()
+        >>> trainers = state.trainers  # HostMesh with 2 nodes
+    """
+
+    def __init__(
+        self,
+        meshes: Dict[str, int],
+        resources: Optional["sky.Resources"] = None,
+        cluster_name: Optional[str] = None,
+        monarch_port: int = DEFAULT_MONARCH_PORT,
+        idle_minutes_to_autostop: Optional[int] = None,
+        down_on_autostop: bool = False,
+        python_exe: str = "python",
+        setup_commands: Optional[str] = None,
+        workdir: Optional[str] = None,
+        file_mounts: Optional[Dict[str, str]] = None,
+    ) -> None:
+        """
+        Args:
+            meshes: Dictionary mapping mesh names to number of nodes.
+                    e.g., {"trainers": 4, "dataloaders": 2}
+            resources: SkyPilot Resources specification for the instances.
+                       If None, uses SkyPilot defaults.
+            cluster_name: Name for the SkyPilot cluster. If None, auto-generated.
+            monarch_port: Port for TCP communication between Monarch workers.
```

Comment:
clarify this is for bootstrap port (its not eh only port for all monarch TCP comm)